Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -250,6 +263,52 @@ public void TestEnumerateDirectory()
                 expectedResult: true);
         }
 
+        [Feature(Features.Symlink)]
+        [Feature(Features.OpaqueDirectory)]
+        [TheoryIfSupported(requiresWindowsBasedOperatingSystem: true, requiresSymlinkPermission: true)]
+        [InlineData(SealDirectoryKind.Full)]
+        [InlineData(SealDirectoryKind.Partial)]
+        [InlineData(SealDirectoryKind.Opaque)]
+        [InlineData(SealDirectoryKind.SharedOpaque)]
+        public void DirectorySymlinksInOutputDirectoryProducerOnlyOnWindows(SealDirectoryKind dirKind)
+        {
+            XAssert.IsFalse(dirKind.IsSourceSeal());
+            
+            AbsolutePath rootDirAbsPath = CreateUniqueObjPath($"{dirKind}.framework");
+            string rootDir = rootDirAbsPath.ToString(Context.PathTable);
+
+            // schedule producer pip: produce opaque directory with a bunch of symlinks
+            var operations = GetLayoutProducingOperationsWithDummyReadFile(rootDir, "producer");
+            var producerPipBuilder = CreatePipBuilder(operations);
+            producerPipBuilder.ToolDescription = StringId.Create(Context.StringTable, "producer");
+
+            // if dirKind is an opaque output, then just declare the root directory as an opaque output;
+            // otherwise, add all outputs explicitly and schedule a seal directory pip to seal them.
+            Process producerPip;
+            DirectoryArtifact outputDirArtifact;
+            if (dirKind.IsOpaqueOutput())
+            {
+                producerPipBuilder.AddOutputDirectory(rootDirAbsPath, kind: dirKind);
+                producerPip = SchedulePipBuilder(producerPipBuilder).Process;
+                outputDirArtifact = producerPip.DirectoryOutputs.First();
+            }
+            else
+            {
+                var dao = InferIOFromOperations(operations, force: true);
+                foreach (var output in dao.Outputs) producerPipBuilder.AddOutputFile(output.Path);
+                foreach (var input in dao.Dependencies) producerPipBuilder.AddInputFile(input);
+                producerPip = SchedulePipBuilder(producerPipBuilder).Process;
+                outputDirArtifact = SealDirectory(rootDirAbsPath, dirKind, dao.Outputs.ToArray());
+            }
+
+            // first run, expect cache miss
+            RunScheduler().AssertSuccess().AssertCacheMiss(producerPip.PipId);
+
+            // rerun, check cache miss, symlink directory in output can't be cached
+            RunScheduler().AssertSuccess().AssertCacheMiss(producerPip.PipId);
+            AssertWarningEventLogged(LogEventId.StorageSymlinkDirInOutputDirectoryWarning, count: 4);          

```

Comment:
Is it true that this warning happens after the first run too?  I'd say yes, so assert it after the first run as well.

Reply:
sure, I will separate them.